pandas/core/frame.py

```diff
@@ -2956,9 +2955,81 @@ def reindex_axis(self, labels, axis=0, method=None, level=None, copy=True,
                                         method=method, level=level, copy=copy,
                                         limit=limit, fill_value=fill_value)
 
-    @Appender(_shared_docs['rename'] % _shared_doc_kwargs)
-    def rename(self, mapper=None, index=None, columns=None, axis=0,
+    def rename(self, mapper=None, index=None, columns=None, axis=None,
                **kwargs):
+        """Alter axes labels.
+
+        Function / dict values must be unique (1-to-1). Labels not contained in
+        a dict / Series will be left as-is. Extra labels listed don't throw an
+        error.
+
+        See the :ref:`user guide <basics.rename>` for more.
+
+        Parameters
+        ----------
+        mapper : dict-like or function, optional
+            Applied to the axis specified by ``axis``
+        index, columns : scalar, list-like, dict-like or function, optional
```

Comment:
'scalar' is not correct anymore ? (as this is only for series) 

also list-like is here and not for `mapper`. Although I am not sure when you actually can use a list-like

BTW, since the type and explanation of both mapper and index/columns are the same, we could also combine all three of them like now index, columns are combined. 
Then inside the description can explain that either axis needs to be specified or either index/columns can be used.